friends/src/App.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import './App.css';
+import FriendsList from './components/FriendsList.js';
+import FriendForm from './components/FriendForm.js';
+import UpdateForm from './components/UpdateForm';
+import axios from 'axios';
+import { Route } from 'react-router-dom';
+
+class App extends React.Component {
+  
+  state ={
+    list: [],
+    friend: {
+      name: "",
+      age: "",
+      email: ""
+    },
+    updatedFriend: {
+      name: "",
+      age: "",
+      email: ""
+    }
+  }
+
+  componentDidMount() {
+    axios
+      .get('http://localhost:5000/friends')
+      .then(response => {
+        this.setState({list : response.data })})
+      .catch(err => console.log(err));
+  }
+
+  handleChanges = e => {
+    e.preventDefault();
+    this.setState({
+      friend: {
+        ...this.state.friend,
+        [e.target.name]: e.target.value
+      },
+      updatedFriend: {
+        ...this.state.updatedFriend,
+        [e.target.name]: e.target.value
+      }
+    })
+  }
+
+  addNewFriend = e => {
+    e.preventDefault();    
+    const newFriend = {
+      name: this.state.friend.name,
+      age: this.state.friend.age,
+      email: this.state.friend.email
+    }
+    axios
+      .post('http://localhost:5000/friends', newFriend)
+      .then(res => {
+        this.setState({list: res.data})
+        this.props.history.push('/')
+      })
+      .catch(err => console.log(err))
+
+    this.setState({
+      friend: {
+        name: "",
+        age: "",
+        email: ""
+      }
+    })
+  }
+
+  updateFriend = (e, id, friend) => {
+    e.preventDefault();    
+    axios
+      .put(`http://localhost:5000/friends/${id}`, friend)
+      .then(res => {
+        this.setState({list: res.data})
+        this.props.history.push('/')
+      })
```

Comment:
Great job accounting for the user experience and rerouting them to the page that is rendering data.